### ISSUE TITLE
[Machine Learning] Hyperparameter optimization for XGBoost algorithm.

### ISSUE THREAD

**Issue by donnexwafula33** [section-engineering-education/engineering-education#6856]
#### Proposed title of article
[Machine Learning] Hyperparameter optimization for XGBoost algorithm.

#### Proposed article introduction
A hyperparameter is a parameter that is set before the learning process begins. These parameters are tunable and can directly affect how well the model trains. Some examples of hyperparameters in machine learning: Learning Rate, number of Epochs, momentum, regularization constant, number of branches in a decision tree, and number of clusters in a clustering algorithm.

Some common strategies for optimizing hyperparameters: Grid Search: Search a set of manually predefined hyperparameters for the best performing hyperparameter.  Random Search: Similar to grid search, but replaces the exhaustive search with random search. Bayesian Optimization: Builds a probabilistic model of the function mapping from hyperparameter values to the target evaluated on a validation set.  Gradient-Based Optimization: Compute gradient using hyperparameters and then optimize hyperparameters using gradient descent. Evolutionary Optimization: Uses evolutionary algorithms (e.g. genetic functions) to search the space of possible hyperparameters.

XGBoost focuses on speed, flexibility, and model performance. It provides a large range of hyperparameters. We can leverage the maximum power of XGBoost by tuning its hyperparameters. XGBoost hyperparameters have been divided into 4 categories. They are as follows: General parameters, booster parameters, learning task parameters, command line parameters. We will fine-tune these parameters so that we have an optimized model with the best results.

#### Key takeaways 
1. Parameter vs hyperparameter.
2. Common strategies for optimizing hyperparameters (Grid search, random search, Optuna, Bayesian optimization, Evolutionary optimization)
3. Categories of XGBoost hyperparameters (General parameters, Booster parameters, Learning task parameters, and Command line parameters)
4. Optimizing the XGBoost hyperparameters in Python.

#### Article quality
This article explains the common strategies for optimizing hyperparameters (Grid search, random search, Optuna, Bayesian optimization, Evolutionary optimization) in detail. A reader will be able to differentiate these strategies. We will also explore all the categories of  XGBoost hyperparameters. We will list all the XGBoost hyperparameters and explain how we can fine-tune them to give the best results. Finally, we will implement the random search and Bayesian optimization technique to optimize a customer churn model.
#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

**Comment by WanjaMIKE** [comment]
Based on this [search](https://www.google.com/search?q=Hyperparameter+optimization+for+XGBoost+algorithm&oq=Hyperparameter+optimization+for+XGBoost+algorithm&aqs=chrome..69i57j69i61j69i60l2.337j0j7&sourceid=chrome&ie=UTF-8), this topic appears a bit over-saturated. We focus more on unique projects. Feel free to suggest another topic. Thanks
